include/sys_clock.h

```diff
@@ -82,7 +82,11 @@ typedef struct {
 #define K_TIMEOUT_EQ(a, b) ((a).ticks == (b).ticks)
 
 #define Z_TIMEOUT_NO_WAIT ((k_timeout_t) {})
+#ifdef __cplusplus
+#define Z_TIMEOUT_TICKS(t) ((k_timeout_t) { (t) })
```

Comment:
I'm having a very hard time coming up with a situation where this should be necessary.  The correct initialization of the structure literal for both C and C++ is the one below.  This isn't a list.

Can you point to a call site in-tree that Eclipse complains about?

Reply:
Thanks @alexanderwachter ; I'd forgotten older versions of C++ didn't support designated initializers.

Then yes, this is correct, but a better condition might be:

```
#if defined(__cplusplus) && ((__cplusplus - 0) < 202002L)
```

Could you try that, please?

Reply:
As expectected it works without problems as well.
Unfortunately there are some weird compliance warnings I have to figure out where is the problem.

Reply:
Thanks.

Those compliance problems are supposed to have been fixed (@nashif), maybe they broke due to #31446.  They're nothing to do with this PR.

Reply:
Ah, no,  #31327 says that you need to rebase on current master, since the compliance infrastructure wasn't doing that for you.

Reply:
Thanks a lot for pointing out this issue.
Done.